src/tests/integration/MetaMorpho.t.sol

```diff
@@ -0,0 +1,479 @@
+        uint256 spDaiTotalSupplyBefore = spDai.totalSupply();
+        uint256 spDaiTotalAssetsBefore = spDai.totalAssets();
+        uint256 spDaiMaxDepositBefore = spDai.maxDeposit(address(pool));
+        uint256 morphoBalanceBefore = dai.balanceOf(address(morpho));
+        uint256 daiTotalSupplyBefore = dai.totalSupply();
+
+        assertEq(spDai.totalSupply(), 0);
+        assertEq(spDai.totalAssets(), 0);
+        assertEq(plan.targetAssets(), 0);
+        assertEq(spDai.maxDeposit(address(pool)), type(uint184).max);
+
+        // Set target assets at `d3mDeposit` and exec.
+        vm.prank(operator);
+        plan.setTargetAssets(d3mDeposit);
+        hub.exec(ilk);
+
+        assertEq(plan.targetAssets(), d3mDeposit);
+
+        assertEq(morpho.market(marketParams.id()).totalSupplyAssets, marketSupplyBefore + d3mDeposit);
+
+        assertEq(spDai.balanceOf(address(pool)), d3mDeposit);
+        assertEq(spDai.totalSupply(), spDaiTotalSupplyBefore + d3mDeposit);
+        assertEq(
+            spDai.totalAssets(), spDaiTotalAssetsBefore + morpho.expectedSupplyAssets(marketParams, address(spDai))
+        );
+        assertEq(spDai.maxDeposit(address(pool)), spDaiMaxDepositBefore - d3mDeposit);
+        assertEq(pool.assetBalance(), UtilsLib.zeroFloorSub(d3mDeposit, 1));
+
+        assertEq(dai.balanceOf(address(morpho)), morphoBalanceBefore + d3mDeposit);
+        assertEq(dai.totalSupply(), daiTotalSupplyBefore + d3mDeposit);
+    }
+
+    function testDepositInOneMarketMoreThanLine(uint256 d3mDeposit) public {
+        d3mDeposit = bound(d3mDeposit, maxLineScaled, type(uint256).max);
+
+        uint256 marketSupplyBefore = morpho.market(marketParams.id()).totalSupplyAssets;
+        uint256 spDaiTotalSupplyBefore = spDai.totalSupply();
+        uint256 spDaiTotalAssetsBefore = spDai.totalAssets();
+        uint256 spDaiMaxDepositBefore = spDai.maxDeposit(address(pool));
+        uint256 morphoBalanceBefore = dai.balanceOf(address(morpho));
+        uint256 daiTotalSupplyBefore = dai.totalSupply();
+
+        assertEq(spDai.totalSupply(), 0);
+        assertEq(spDai.totalAssets(), 0);
+        assertEq(plan.targetAssets(), 0);
+        assertEq(spDai.maxDeposit(address(pool)), type(uint184).max);
+
+        // Set target assets at `d3mDeposit` and exec.
+        vm.prank(operator);
+        plan.setTargetAssets(d3mDeposit);
+        hub.exec(ilk);
+
+        assertEq(plan.targetAssets(), d3mDeposit);
+
+        assertEq(morpho.market(marketParams.id()).totalSupplyAssets, marketSupplyBefore + maxLineScaled);
+
+        assertEq(spDai.balanceOf(address(pool)), maxLineScaled);
+        assertEq(spDai.totalSupply(), spDaiTotalSupplyBefore + maxLineScaled);
+        assertEq(
+            spDai.totalAssets(), spDaiTotalAssetsBefore + morpho.expectedSupplyAssets(marketParams, address(spDai))
+        );
+        assertEq(spDai.maxDeposit(address(pool)), spDaiMaxDepositBefore - maxLineScaled);
```

Comment:
Any reason to not have added the same check than the prev two:
```
assertEq(pool.assetBalance(), UtilsLib.zeroFloorSub(d3mDeposit, 1));
```

Reply:
fixed in 576c75cb8a76dbbd3390335920307bd3f09b7647